internal/agent/upgrade.go

```diff
@@ -148,17 +95,133 @@ func determineUpgradeImage(version string) (string, error) {
 		Version: version,
 	})
 	if err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if img != "" {
-		return img, nil
+		return nil, nil
 	}
 
 	registry, err := utils.OSRelease("IMAGE_REPO")
 	if err != nil {
-		return "", fmt.Errorf("can't find IMAGE_REPO key under /etc/os-release %w", err)
+		return nil, fmt.Errorf("can't find IMAGE_REPO key under /etc/os-release %w", err)
+	}
+
+	return v1.NewSrcFromURI(fmt.Sprintf("%s:%s", registry, version))
+}
+
+// generateUpgradeConfForCLIArgs creates a kairos configuration for `--source` and `--recovery`
+// command line arguments. It will be added to the rest of the configurations.
+func generateUpgradeConfForCLIArgs(source string, upgradeRecovery bool) (string, error) {
```

Comment:
same comment, though I see why source might have been prefered, though I still think in functions it's worth the detail of what that source represents

Reply:
Yes "source" was chosen because it matches the command line argument exactly: `--source`. I can rename the functions though.

Reply:
renamed to `sourceImageURL`